ReadingRainbowAPI/Controllers/BookController.cs

```diff
@@ -90,6 +90,38 @@ public async Task<IActionResult> AddBooksToWishList(string userName, Book book)
             return Ok();
         }
 
+        [HttpPost]
+        [Route("RemoveBookFromLibrary/{userName}")]
+        public async Task<IActionResult> RemoveBookFromLibrary(string userName, Book book)
+        {
+            await _bookRepository.DeleteBookAsync(book);
+
+            foreach(var genre in book.Genres)
+            {
+                await _genreRepository.DeleteInGenreRelationshipAsync(genre, book, new Relationships.InGenre());
+            }
+
+            await _bookRepository.DeleteInLibraryRelationshipAsync(book, new Person() {Name = userName}, new InLibrary());

```

Comment:
You probably only need to delete the relationship here? The book could be in other user's libraries/wish lists and those relationships should be preserved. After you delete the in Library relationship you could check if there are still any people associated with the book (by using the GetInLibraryPersonRelationshipAsync function in the book repository) and deleting if it returns no people.